pkg/plugins/policies/meshaccesslog/plugin/v1alpha1/plugin.go

```diff
@@ -145,9 +158,50 @@ func applyToDirectAccess(
 	return nil
 }
 
+func applyToGateway(
+	rules xds.ToRules, gatewayListeners map[xds.InboundListener]*envoy_listener.Listener, resources xds_context.ResourceMap, dataplane *core_mesh.DataplaneResource,
+) error {
+	var gateways *core_mesh.MeshGatewayResourceList
+	if rawList := resources[core_mesh.MeshGatewayType]; rawList != nil {
+		gateways = rawList.(*core_mesh.MeshGatewayResourceList)
+	} else {
+		return nil
+	}
+
+	gateway := xds_topology.SelectGateway(gateways.Items, dataplane.Spec.Matches)
+	if gateway == nil {
+		return nil
+	}
```

Comment:
what do you think about enriching `Proxy` object and moving this logic to `DataplaneProxyBuilder`?

Reply:
Definitely, fixing `Proxy` has been on my list for a while

Reply:
https://github.com/kumahq/kuma/issues/5124